src/app/rundown-view/services/action-trigger-producer-key-binding.service.ts

```diff
@@ -94,26 +96,16 @@ export class ActionTriggerProducerKeyBindingService implements KeyBindingService
     })
   }
 
-  private emitNewKeybindings(): void {
-    this.keyBindings = this.createKeyBindings()
-    this.keyBindingsSubject.next(this.keyBindings)
-  }
-
-  private createKeyBindings(): KeyBinding[] {
-    if (!this.rundown) {
-      return []
-    }
-
-    const keyBindings: StyledKeyBinding[] = this.keyBindingFactory.createRundownKeyBindings(this.rundown)
-
-    if (this.rundown.mode === RundownMode.INACTIVE) {
-      return keyBindings
-    }
-
+  private updateActionTriggerKeybindings(): void {
+    const keyBindings: KeyBinding[] = []
     this.actionTriggersWithAction.forEach((action: Tv2Action, actionTrigger: ActionTrigger<KeyboardTriggerData>) => {
       keyBindings.push(this.createBinding(action, actionTrigger, this.rundown!.id))
     })
-    return keyBindings
+    this.actionTriggerKeyBindings = keyBindings
```

Comment:
The complexity of this can be reduced with by making this into a single expression either with `Object.entries`, `[...mapVariable]` or `Array.from`. I have provided a suggestion with `Array.from`.
```typescript
this.actionTriggerKeyBindings = Array.from(
  this.actionTriggersWithAction,
  ([actionTrigger, action]) => this.createBinding(action, actionTrigger, this.rundown!.id)
)
```

Reply:
I wasn't aware of the override of `Array.from()`. I'm a fan :) 
It has been updated.